src/arch/xtensa/configs/cannonlake_gcc_defconfig

```diff
@@ -3,6 +3,6 @@ CONFIG_INTEL_DMIC=y
 CONFIG_INTEL_SSP=y
 CONFIG_INTEL_ALH=y
 CONFIG_CAVS_LPS=y
-CONFIG_CORE_COUNT=4
+CONFIG_CORE_COUNT=1
```

Comment:
Should we have an error/check here since today we can have CONFIG_MULTICORE=y and CORE_COUNT=1
it may be better to have a #error e.g.
```c
#ifdef __ZEPHYR__
#if CORE_COUNT > 1
#error blah blah blah
#endif 
#endif 
```
This ways its obvious core count must be 1 and it can be merged (and later reverted).

Reply:
There are several configuration options, related to SMP support, used in SOF under Zephyr: `CONFIG_CORE_COUNT`, `CONFIG_SMP`, `CONFIG_MP_NUM_CPUS`, and then of course there's also `CONFIG_MULTICORE` in SOF itself. To make matters even worse they're "defined" in multiple places. E.g. in zephyr/kernel/Kconfig
```
config SMP
	bool "Enable symmetric multithreading support"
	depends on USE_SWITCH
	help
	  When true, kernel will be built with SMP support, allowing
	  more than one CPU to schedule Zephyr tasks at a time.

config MP_NUM_CPUS
	int "Number of CPUs/cores"
	default 1
	range 1 4
	help
	  Number of multiprocessing-capable cores available to the
	  multicpu API and SMP features.
```

and then, e.g. in zephyr/soc/xtensa/intel_adsp/cavs_v18/Kconfig.defconfig.series
```
config SMP
       default y

config MP_NUM_CPUS
	default 2
```

I tried to average out all those options a bit, the result is in https://github.com/andyross/zephyr/pull/1/commits/4d7f6ab34ce537ad9b77adab2b164293dd86fdeb So, I'm not really sure which config option is the "top level" multi-core support that we should be checking for?

Reply:
ok, these need aligned to use the same as Zephyr. Can we have a mergable build error for now and we can fix properly with @andyross for Zephyr 2.5 and SOF v1.7